src/software/gui/full_system/threaded_full_system_gui.cpp

```diff
@@ -22,6 +22,10 @@ ThreadedFullSystemGUI::ThreadedFullSystemGUI()
           std::make_shared<ThreadSafeBuffer<SensorProto>>(SENSOR_MSG_BUFFER_SIZE)),
       view_area_buffer(
           std::make_shared<ThreadSafeBuffer<Rectangle>>(VIEW_AREA_BUFFER_SIZE, false)),
+      data_received_per_second_buffer(
+          std::make_shared<ThreadSafeBuffer<double>>(DATA_PER_SECOND_BUFFER_SIZE, false)),
+      data_sent_per_second_buffer(
+          std::make_shared<ThreadSafeBuffer<double>>(DATA_PER_SECOND_BUFFER_SIZE, false)),
```

Comment:
I think we can call the `data_sent_per_second_buffer` -> `primitive_sets_sent_per_second_buffer` and `data_received_per_second_buffer` -> `world_received_per_second_buffer` and update that in everything to do w/ specific msgs. 

On the visualizer, I think we should have `Primitives Sent Per Second` and `World Received Per Second` instead of `Data Sent Per Second` and `Data Received Per Second` just to be more clear on what the data actually is.